### DISCUSSION TITLE
Optimizing CPU Usage for Multiple CefSharp Instances Running JavaScript Canvas-Based Web Games

### DISCUSSION THREAD

**Discussion by HaukHuang** [cefsharp/CefSharp#4884] [General]
Hi everyone,

I'm currently using CefSharp to run a web game that is implemented using JavaScript and HTML5 canvas. Each instance of the game consumes about 15% of CPU, and since I need to run multiple instances simultaneously, I’m looking for ways to reduce CPU usage as much as possible.

I have already tried setting various command line parameters, but I haven’t observed a significant reduction in CPU usage. I’m wondering if there are additional methods to minimize CPU consumption, such as disabling certain browser features or optimizing CefSharp settings further.

Has anyone faced similar issues or found effective strategies for reducing CPU usage in this scenario? Any advice or insights would be greatly appreciated!

Thank you!

**Comment by HaukHuang** [comment]
Is it possible to reduce CPU usage by lowering the graphics quality or frame rate?

**Comment by amaitland** [comment]
Make sure your application as an `app.manifest` with `Windows 10` compatibility entry so GPU detection works correctly.

https://github.com/cefsharp/CefSharp/wiki/Quick-Start-For-MS-.Net-4.x#40-add-appmanifest-to-your-application

Load `chrome://gpu` in ` ChromiumWebBrowser` instance and in `Chrome` to compare.

> Is it possible to reduce CPU usage by lowering the graphics quality or frame rate?

You can change the frame rate, depends if you are using WPF or WinForms on how you do it.

If you are using the `WPF` implementation then switching to https://github.com/cefsharp/CefSharp.Wpf.HwndHost will give better performance.

**Reply by HaukHuang** [reply]
I am using WinForms. How can I reduce the frame rate?
----Thanks
